app/components/data-download.tsx

```diff
@@ -98,43 +62,162 @@ const DataDownloadInner = memo(
-        )}
-      </DownloadButton>
+      <>
+        <DownloadButton iconName="download" {...bindHover(popupState)}>
+          <Trans id="button.download">Download</Trans>
+        </DownloadButton>
+        <HoverMenu
+          {...bindMenu(popupState)}
+          anchorOrigin={{ vertical: "top", horizontal: "center" }}
+          transformOrigin={{ vertical: "bottom", horizontal: "center" }}
+        >
+          <Typography
+            sx={{
+              ...PADDING_PROPS,
+              borderBottom: "1px solid",
+              borderBottomColor: "grey.500",
+            }}
+          >
+            Download
+          </Typography>
+          {OPTIONS_TO_RENDER.map((d) => (
+            <DownloadMenuItem
+              key={d.extent + d.fileFormat}
+              title={title}
+              extent={d.extent}
+              fileFormat={d.fileFormat}
+              closeMenu={popupState.close}
+              {...(d.extent === "all" ? allPrepared : visiblePrepared)}
+            />
+          ))}
+        </HoverMenu>
+      </>
     );
   }
 );
 
-const RunSparqlQuery = ({
-  url,
+const DownloadMenuItem = ({
+  title,
+  data,
+  columnKeys,
   extent,
+  fileFormat,
+  closeMenu,
 }: {
-  url: string;
-  extent: DataDownloadExtent;
+  title: string;
+  data: Observation[];
+  columnKeys: string[];
+  extent: Extent;
+  fileFormat: FileFormat;
+  closeMenu: () => void;
+}) => {
+  const download = useCallback(() => {
+    return fetch("/api/download", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        columnKeys,
+        data,
+        fileFormat,
+      }),
+    }).then((res) =>
+      res.blob().then((blob) => saveAs(blob, `${title}.${fileFormat}`))
+    );
+  }, [columnKeys, data, fileFormat, title]);
+
+  return (
+    <DownloadMenuItemInner onClick={() => download().then(() => closeMenu())}>
+      {extent === "visible" ? (
+        <>
+          <Trans id="button.download.data.visible">Overlooking dataset</Trans> (
+          {fileFormat.toUpperCase()})
+        </>
+      ) : (
+        <>
+          <Trans id="button.download.data.all">Full dataset</Trans> (
+          {fileFormat.toUpperCase()})
+        </>
+      )}
+    </DownloadMenuItemInner>
+  );
+};
+
+const DownloadMenuItemInner = ({
+  children,
+  onClick,
+}: {
+  children: ReactNode;
+  onClick: () => void;
 }) => {
+  return (
+    <MenuItem onClick={onClick} sx={{ ...PADDING_PROPS }}>
+      <DownloadButton iconName="table">{children}</DownloadButton>
+    </MenuItem>
+  );
+};
+
+const DownloadButton = ({
+  children,
+  iconName,
+  ...props
+}: {
+  children: ReactNode;
+  iconName: IconName;
+}) => {
+  return (
+    <Button
+      component="a"
+      variant="text"
+      color="primary"
+      size="small"
+      startIcon={<Icon name={iconName} size={16} />}
+      {...props}
+      sx={{ fontWeight: "regular", ":hover": { color: "primary.main" } }}
```

Comment:
Why do we need to pass sx with hover here, is the style really that different from "variant=text" or should we align variant='text" theme ? Better to add a typography variant here if necessary here ? We shuld not have in general to pass sx for customisations (exception would be for padding/margin override).

Reply:
This is due to the fact that MenuItem has its own onHover styling; if I left the default styling for this button, then there would be two onHover styles, one in general for MenuItem, and second one if you hover over the text (color would change).  I think I can also replace this button with just typography + icon, I used it because it's the same as the button to open the menu, but it also will be possible to have the same styling with other elements.

Reply:
Yes, I think it is unnecessary to reuse DownloadButton inside the MenuItem, since it complicates things = we have to cancel out the style. Better to only use `<><ListItemIcon><Icon name={iconName} size={16} /></ListItemIcon><ListItemText>Download</ListItemText></>` IMHO.